### ISSUE TITLE
Regexes not working for `accept` configuration settings

### ISSUE THREAD

**Issue by troessner** [troessner/reek#888]
Working on #886 and #887 uncovered another bug in our configuration handling (* sigh *). 
Using this feature works:

```cucumber
Feature: `accept` configuration setting
  In order to have a more fine-grained control over what Reek reports
  As a user
  I want to be able to accept specific pattern and exclude them from reporting

  Scenario: Exclude multiple patterns
    Given a file named "config.reek" with:
      """
      ---
      UncommunicativeMethodName:
        accept:
          - m1
          - m2
      UncommunicativeParameterName:
        accept:
          - a1
          - a2
      UncommunicativeModuleName:
        accept:
          - C1
      """
    And a file named "smelly.rb" with:
      """
      # Should not report UncommunicativeModuleName
      class C1
        # Should not report UncommunicativeMethodName and UncommunicativeParameterName
        def m1(a1); a1; end
        # Should not report UncommunicativeMethodName and UncommunicativeParameterName
        def m2(a2); a2; end
        # Should report UncommunicativeMethodName and UncommunicativeParameterName
        def m3(a3); a3; end
      end
      """
    When I run `reek -c config.reek smelly.rb`
    Then it reports:
    """
    smelly.rb -- 2 warnings:
      [8]:UncommunicativeMethodName: C1#m3 has the name 'm3' [https://github.com/troessner/reek/blob/master/docs/Uncommunicative-Method-Name.md]
      [8]:UncommunicativeParameterName: C1#m3 has the parameter name 'a3' [https://github.com/troessner/reek/blob/master/docs/Uncommunicative-Parameter-Name.md]
    """
```

But not when I use regexes likes this:

```yaml
      ---
      UncommunicativeMethodName:
        accept:
          - /^m1$/
```

According to [our docs](https://github.com/troessner/reek/blob/master/docs/Uncommunicative-Method-Name.md#configuration) this should work.
I suspect that this bug has been in Reek forever (since I can remember any of us touching the `accept` setting for years) but nobody noticed or reported it.

**Comment by mvz** [comment]
@troessner is this still an issue?

**Comment by troessner** [comment]
Nope, fixed with #890 -> closed!
